frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -768,6 +781,365 @@ class SegmentsView extends React.Component<Props, State> {
+              true,
+            );
+            this.handleSegmentDropdownMenuVisibility(groupId, false);
+          },
+          disabled: !this.props.allowUpdate,
+          icon: <ArrowRightOutlined />,
+          label: "Move active segment here",
+        }
+      : null;
+  };
+
+  getShowMeshesItem = (groupId: number): ItemType => {
+    return this.state != null && this.doesGroupHaveAnyMeshes(groupId)
+      ? {
+          key: "showMeshesOfGroup",
+          label: (
+            <div
+              onClick={() => {
+                if (this.props.visibleSegmentationLayer == null) {
+                  // Satisfy TS
+                  return;
+                }
+                this.handleChangeMeshVisibility(
+                  this.props.visibleSegmentationLayer.name,
+                  groupId,
+                  !this.state.areSegmentsInGroupVisible[groupId],
+                );
+                this.handleSegmentDropdownMenuVisibility(groupId, false);
+              }}
+            >
+              <i className="fas fa-dice-d20" />
+              {this.state.areSegmentsInGroupVisible[groupId] ? "Hide" : "Show"} all meshes
+            </div>
+          ),
+        }
+      : null;
+  };
+
+  setGroupColor(groupId: number, color: Vector3) {
+    const { visibleSegmentationLayer } = this.props;
+    if (visibleSegmentationLayer == null) return;
+    const segmentGroupToChangeColor = this.getSegmentsOfGroup(groupId);
+    if (segmentGroupToChangeColor == null) return;
+
+    const actionArray = segmentGroupToChangeColor.map((segment) =>
+      updateSegmentAction(segment.id, { color: color }, visibleSegmentationLayer.name),
+    );
+
+    Store.dispatch(batchUpdateGroupsAndSegmentsAction(actionArray));
+  }
+
+  handleRefreshMeshes = (groupId: number) => {
+    const { visibleSegmentationLayer } = this.props;
+    if (visibleSegmentationLayer == null) return;
+    const segmentGroup = this.getSegmentsOfGroup(groupId);
+    if (segmentGroup == null) return;
+
+    segmentGroup.forEach((segment) => {
+      if (
+        Store.getState().localSegmentationData[visibleSegmentationLayer.name].isosurfaces[
+          segment.id
+        ] != null
+      ) {
+        Store.dispatch(refreshIsosurfaceAction(visibleSegmentationLayer.name, segment.id));
+      }
+    });
+  };
+
+  handleRemoveMeshes = (groupId: number) => {
+    const { visibleSegmentationLayer } = this.props;
+    if (visibleSegmentationLayer == null) return;
+    const segmentGroup = this.getSegmentsOfGroup(groupId);
+    if (segmentGroup == null) return;
+
+    segmentGroup.forEach((segment) => {
+      if (
+        Store.getState().localSegmentationData[visibleSegmentationLayer.name].isosurfaces[
+          segment.id
+        ] != null
+      ) {
+        Store.dispatch(removeIsosurfaceAction(visibleSegmentationLayer.name, segment.id));
+      }
+    });
+  };
+
+  handleChangeMeshVisibility = (layerName: string, groupId: number, isVisible: boolean) => {
```

Comment:
the method name should reflect that it details with the meshes within a group. currently, it sounds like it would change the visibility of one mesh.